Add Pylar and Pencil languages support

miguelgargallo opened the pull request:
<!--- Briefly describe your changes in the field above. -->

## Description
<!--- If necessary, go into depth of what this pull request is doing. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- Feel free to remove whole sections, not points within the sections, that do not apply -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      <!-- Replace FOOBAR with the new extension, and KEYWORDS with keywords unique to the language. Repeat for each extension added. -->
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      <!-- Replace FOOBAR with the new extension, and KEYWORDS with keywords unique to the language. Repeat for each extension added. -->
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [x] I have included a syntax highlighting grammar: [URL to grammar repo]
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  - Old: [URL to grammar repo]
  - New: [URL to grammar repo]

- [ ] **I am updating a grammar submodule**
  <!-- That's not necessary, grammar submodules are updated automatically with each new release. -->

- [ ] **I am adding new or changing current functionality**
  <!-- This includes modifying the vendor, documentation, and generated lists. -->
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  <!-- Please ensure you have gathered agreement from the wider language community _before_ opening this PR -->
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

Nixinova commented on lib/linguist/languages.yml:
```diff
@@ -5330,6 +5330,39 @@ Python traceback:
+  - pyl
+  - teide
+  - pyrus
+  extensions:
```
Youve listed `extensions` twice with different contents.

miguelgargallo replied:
.pylar, fixed on next pull request

Nixinova commented on lib/linguist/languages.yml:
```diff
@@ -5330,6 +5330,39 @@ Python traceback:
+  color: "#fed659"
+  ace_mode: text
+  tm_scope: source.pylar
+  language_id: 21
```
```suggestion
  language_id: 383531292
```
The ID should be a hash of the language name.

miguelgargallo replied:
Thanks

lildude replied:
Nope. Everything needs to be fixed in _this_ PR.

Nixinova commented on lib/linguist/languages.yml:
```diff
@@ -7861,6 +7894,33 @@ mcfunction:
+    - example.mdp
+    - example.mdp
+  tm_scope: source.mdp
+  language_id: 178
```
```suggestion
  language_id: 53400606
``` 
ditto

miguelgargallo replied:
53400606, fixed on next pull request

Nixinova commented on lib/linguist/languages.yml:
```diff
@@ -7861,6 +7894,33 @@ mcfunction:
+    - ".pencil"
+    - ".xpro"
+    - ".mdpro"
+  filenames:
+    - style1.mdp
+    - style2.mdp
+    - style3.mdp
+    - example.mdp
+    - example.mdp
```
```suggestion

```
If the extension is `mdp` then you dont need to manually specify any filenames that also have that extension

miguelgargallo replied:
.pencil, fixed on next pull request

lildude commented on vendor/README.md:
```diff
@@ -420,6 +421,7 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
 - **Python:** [tree-sitter/tree-sitter-python](https://github.com/tree-sitter/tree-sitter-python) 🐌
 - **Python console:** [MagicStack/MagicPython](https://github.com/MagicStack/MagicPython)
 - **Python traceback:** [MagicStack/MagicPython](https://github.com/MagicStack/MagicPython)
+- **Pylar:** [miguelgargallo/pylar](https//github.com/miguelgargallo/pylar)
```
This is not a Textmate compatible grammar repo. Please see the CONTRIBUTING.md file.

lildude commented on vendor/licenses/git_submodule/mdp.yml:
```diff
@@ -0,0 +1,492 @@
+---
```
This is not a valid license file. Please see the CONTRIBUTING.md file.

miguelgargallo replied:
✅ Done, Added MIT 2023 and Apache 2.0

lildude commented on vendor/licenses/git_submodule/pylar.yml:
```diff
@@ -0,0 +1,492 @@
+---
```
This is not a valid license file. Please see the CONTRIBUTING.md file.

miguelgargallo replied:
✅ Done, Added MIT 2023 and Apache 2.0

lildude reviewed (changes requested):
🤔 You have clearly not followed the instructions in the CONTRIBUTING.md file and have chosen to do things manually. Please don't. I can tell as you have made manual changes to the `.gitmodules`, `grammars.yml` and `vendor/README.md` files and have manually added the wrong license files (these should be of the grammar(s) used for the syntax highlighting). These files are added/maintained by the scripts in the repo so do not need manual modifications.

Please revert all of your changes (or close and start again) and follow the instructions in the CONTRIBUTING.md file.

Please update the OP with a link to the search results for each of the extensions and filenames you are proposing. Keep in mind that we have minimum usage requirements (once again, see CONTRIBUTING.md) so please do not include any that do not meet these requirements.

Please also update the OP with links to the source of all the samples you're adding and specify the license for each.

lildude commented:
Oh yes, and you only need to fill in the "I am adding a new language." section of the template in the OP. You can remove the other sections.

miguelgargallo reviewed (commented):
All correct folowwing the guide

miguelgargallo commented on vendor/README.md:
```diff
@@ -311,6 +311,7 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
 - **Makefile:** [textmate/make.tmbundle](https://github.com/textmate/make.tmbundle)
 - **Mako:** [marconi/mako-tmbundle](https://github.com/marconi/mako-tmbundle)
 - **Markdown:** [atom/language-gfm](https://github.com/atom/language-gfm)
+- **MarkdownPylar:** [miguelgargallo/mdp](https://github.com/miguelgargallo/mdp)
```
✅ Done, passed the docker command script of grammar

miguelgargallo commented:
> 🤔 You have clearly not followed the instructions in the CONTRIBUTING.md file and have chosen to do things manually. Please don't. I can tell as you have made manual changes to the `.gitmodules`, `grammars.yml` and `vendor/README.md` files and have manually added the wrong license files (these should be of the grammar(s) used for the syntax highlighting). These files are added/maintained by the scripts in the repo so do not need manual modifications.
> 
> Please revert all of your changes (or close and start again) and follow the instructions in the CONTRIBUTING.md file.
> 
> Please update the OP with a link to the search results for each of the extensions and filenames you are proposing. Keep in mind that we have minimum usage requirements (once again, see CONTRIBUTING.md) so please do not include any that do not meet these requirements.
> 
> Please also update the OP with links to the source of all the samples you're adding and specify the license for each.

Hi there,

Thank you for bringing these issues to our attention. We apologize for not following the instructions in the CONTRIBUTING.md file and for making manual changes to the .gitmodules, grammars.yml, and vendor/README.md files. We have now reverted all of our changes and are following the instructions in the CONTRIBUTING.md file.

We have updated the OP with links to the search results for each of the extensions and filenames we are proposing, and have included only those that meet the minimum usage requirements. We have also included links to the source of all the samples we are adding and have specified the license for each.

We hope this resolves the issue. Please let us know if there is anything else we can do to assist.

Thank you for your understanding.

Best regards,
Miguel Gargallo

lildude reviewed (changes requested):
Please disable your editor's auto-formatter and revert the formatting changes you've made to the `languages.yml` file. We only want changes for the entries you're adding.

Please also remove all the `.DS_Store` files.

Please also remove all the duplicate samples. Multiple samples of the same content don't help the classifier at all.

Your samples also don't look like real world examples and you've even included an "hello world" sample. We only want real world samples, not contrived examples as they're used to train the classifier.

Please remove the `mdp.yml` file too.

> We have updated the OP with links to the search results for each of the extensions and filenames we are proposing, and have included only those that meet the minimum usage requirements. We have also included links to the source of all the samples we are adding and have specified the license for each.

No you haven't 😉

miguelgargallo commented:
I will work hard to fix this ASAP! Omg 🫢

lildude commented:
There's no rush. It won't be merged until close to making the next release.
